polkadot/node/core/candidate-validation/src/tests.rs

```diff
@@ -480,95 +480,22 @@ fn candidate_validation_bad_return_is_invalid() {
 	let candidate_receipt = CandidateReceipt { descriptor, commitments_hash: Hash::zero() };
 
 	let v = executor::block_on(validate_candidate_exhaustive(
-		MockValidateCandidateBackend::with_hardcoded_result(Err(
+        MockValidateCandidateBackend::with_hardcoded_result(Err(
 			ValidationError::InvalidCandidate(WasmInvalidCandidate::HardTimeout),
 		)),
-		validation_data,
-		validation_code,
-		candidate_receipt,
-		Arc::new(pov),
-		ExecutorParams::default(),
-		PvfExecTimeoutKind::Backing,
-		&Default::default(),
+        validation_data,
+        validation_code,
+        candidate_receipt,
+        Arc::new(pov),
+        ExecutorParams::default(),
+        PvfExecKind::Backing,
+        &Default::default(),
 	))
 	.unwrap();
 
 	assert_matches!(v, ValidationResult::Invalid(InvalidCandidate::Timeout));
 }
 
-// Test that we vote valid if we get `AmbiguousWorkerDeath`, retry, and then succeed.
-#[test]
-fn candidate_validation_one_ambiguous_error_is_valid() {
```

Comment:
Maybe I missed it, but why remove this test? I think we just need to change it from backing -> approval for it to work.